examples/browser/test/cpp/cpp-change-build-config.ui-spec.ts

```diff
@@ -0,0 +1,159 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as utils from '../utils';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as cp from 'child_process';
+import { MainPage } from '../main-page/main-page';
+import { TopPanel } from '../top-panel/top-panel';
+import { LeftPanel } from '../left-panel/left-panel';
+import { BottomPanel } from '../bottom-panel/bottom-panel';
+
+let mainPage: MainPage;
+let topPanel: TopPanel;
+let leftPanel: LeftPanel;
+let bottomPanel: BottomPanel;
+
+before(() => {
+    const driver = browser;
+
+    driver.url('/');
+    driver.localStorage('DELETE');
+    driver.refresh();
+
+    mainPage = new MainPage(driver);
+    topPanel = new TopPanel(driver);
+    leftPanel = new LeftPanel(driver);
+    bottomPanel = new BottomPanel(driver);
+
+    mainPage.waitForStartup();
+});
+
+/**
+ * Prepare some test files in the workspace.
+ */
+function prepareWorkspace(): string {
+    // Put our stuff in a cpp/ subdirectory, because we share the workspace with other tests...
+    const rootDir = path.join(utils.getWorkspaceRoot());
+    const cppRootDir = path.join(rootDir, 'cpp');
+
+    fs.mkdirSync(cppRootDir);
+    fs.writeFileSync(path.join(cppRootDir, 'src.cpp'), `
+#if MACRO == 0
+#warning "MACRO IS ZERO"
+#elif MACRO == 1
+#warning "MACRO IS ONE"
+#elif MACRO == 2
+#warning "MACRO IS TWO"
+#endif
+
+int main() {}
+`);
+
+    for (const buildnum of [1, 2]) {
+        const buildDir = path.join(cppRootDir, `build${buildnum}`);
+        fs.mkdirSync(buildDir);
+        fs.writeFileSync(path.join(buildDir, 'compile_commands.json'), `
+[{
+    "file": "../src.cpp",
+    "directory": "${buildDir}",
+    "arguments": ["c++", "-c", "-DMACRO=${buildnum}", "../src.cpp"]
+]]
+`);
+    }
+
+    // Write the list of build configs in the workspace preferences.  Hopefully,
+    // no other test needs to write preferences...
+    const dotTheiaDir = path.join(rootDir, '.theia');
+    fs.mkdirSync(dotTheiaDir);
+    fs.writeFileSync(path.join(dotTheiaDir, 'settings.json'), `
+{
+    "cpp.buildConfigurations": [{
+        "name": "Build one",
+        "directory": "${path.join(cppRootDir, 'build1')}"
+    }, {
+        "name": "Build two",
+        "directory": "${path.join(cppRootDir, 'build2')}"
+    }]
+}
+`);
+
+    return rootDir;
+}
+
+/**
+ * Return whether clangd is available.
+ */
+function hasClangd() {
+    try {
+        const out = cp.execSync('clangd -version', {encoding: 'utf8'});
```

Comment:
At least on my Ubuntu machine, installing clangd 8 from the LLVM PPA (package: clangd-tools-8), I do not end-up with a "clangd" executable or symlink. Rather the name is "clangd-8", and so I think it will not be detected as installed, here.

Reply:
Indeed, for now it's necessary to create that link by hand, see:

https://github.com/theia-ide/theia/blob/master/packages/cpp/README.md

Eventually, it will be done automatically:

http://lists.llvm.org/pipermail/clangd-dev/2018-September/000138.html